src/app/task/create-task/create-task.component.ts

```diff
@@ -28,8 +28,15 @@ export class CreateTaskComponent {
   constructor(private taskService: TaskService, private router: Router) {}
 
   createTask() {
-    this.newTask = new Task(this.taskForm.value.title, this.taskForm.value.description, this.taskForm.value.type);
-    this.taskService.createTask(this.newTask);
-    this.router.navigate(['/tasks/task-list']);
+    if (this.taskForm.value.title && this.taskForm.value.description && this.taskForm.value.type) {
+      this.newTask = new Task(this.taskForm.value.title, this.taskForm.value.description, this.taskForm.value.type);
+      
+      this.taskService.createTask(this.newTask).subscribe(() => {
```

Comment:
Here you redirect to task list page on either instance, so you can just redirect when task has been created or if task is not valid, you can show an error on the screen